test/pathOrLazy.js

```diff
@@ -0,0 +1,53 @@
+import sinon from 'sinon';
+
+import * as RA from '../src';
+import eq from './shared/eq';
+
+describe('pathOrLazy', function() {
+  const obj = {
+    a: { b: { c: 1, d: ['x', 'y', 'z'] } },
+    e: NaN,
+    f: null,
+    g: undefined,
+  };
+
+  it('returns value for requested path', function() {
+    eq(RA.pathOrLazy(() => {}, ['a', 'b', 'c'], obj), 1);
+  });
+
+  it('returns default value when requested path does not exist', function() {
```

Comment:
If the test description contains conditional logic, they're harder to read. To eliminate complexity we're using [gherkin] (https://www.guru99.com/gherkin-test-cucumber.html) language and nested contexts to eliminate the complexity. Example of this can be found here: https://github.com/char0n/ramda-adjunct/blob/e440acdfa5c7f59475251226e6b0cd3e13d0b59c/test/allEqual.js#L25

The rules are: nested `describe`  will become `context`. Every `it` in `context` will become `specify`.

```js
context('given requested path does not exist', function() {
  specify('should return default value', function() {
    eq(RA.pathOrLazy(() => 7, ['nonexistent'], obj), 7);
  });
});